src/test/java/org/opensearch/security/dlic/rest/api/TenantInfoActionTest.java

```diff
@@ -72,17 +79,78 @@ public void testTenantInfoAPIUpdate() throws Exception {
 
         // update security config
         RestHelper.HttpResponse response = rh.executePatchRequest(
-            BASE_ENDPOINT + "/api/securityconfig",
-            "[{\"op\": \"add\",\"path\": \"/config/dynamic/kibana/opendistro_role\"," + "\"value\": \"opendistro_security_internal\"}]",
-            new Header[0]
-        );
+                BASE_ENDPOINT + "/api/securityconfig",
+                "[{\"op\": \"add\",\"path\": \"/config/dynamic/kibana/opendistro_role\","
+                        + "\"value\": \"opendistro_security_internal\"}]",
+                new Header[0]);
         Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
 
-        response = rh.executePutRequest(BASE_ENDPOINT + "/api/rolesmapping/opendistro_security_internal", payload, new Header[0]);
+        response = rh.executePutRequest(BASE_ENDPOINT + "/api/rolesmapping/opendistro_security_internal", payload,
+                new Header[0]);
         Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
 
         rh.sendAdminCertificate = false;
         response = rh.executeGetRequest(ENDPOINT);
         Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
     }
+
+    @Test
+    public void testParallelPutRequests() throws Exception {
+
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        String[] tenantNames = { "tenant_1", "tenant_2" };
+        String[] descriptions = { "create tenant 1", "create tenant 2" };
+        HttpResponse[] responses = executeMultipleAsyncPutRequest(tenantNames, descriptions);
+        int numCreatedResponses = 0;
+        int numConflictResponses = 0;
+        Integer requestIndexCreated = null;
+        for (int i = 0; i < responses.length; i++) {
+            HttpResponse response = responses[i];
+            if (response.getStatusCode() == HttpStatus.SC_CREATED) {
+                numCreatedResponses++;
+                requestIndexCreated = i;
```

Comment:
wdyt about moving the request and assertion from L124-126 to here to avoid having to keep track of the index of the request with the SC_CREATED response?